crucible-mir/src/Mir/TransTy.hs

```diff
@@ -1449,7 +1481,12 @@ initialValue (M.TyAdt nm _ _) = do
                 Just (discr, var) -> do
                     fldExps <- mapM initField (var^.M.vfields)
                     Just <$> buildEnum' adt discr fldExps
-        M.Union -> return Nothing
+        M.Union ->
+            -- Unions are default-initialized to an untyped `MirAggregate` of an
+            -- appropriate size, like tuples. See Note [union representation] in
+            -- `Mir.Trans`.
+            let unionSize = fromIntegral (adt ^. M.adtSize)
```

Comment:
Needs `unionSize` adjusted as in previous comments

Reply:
Changed to size 1 in 4612fe406b99cb2a11258199828875e39f447cd1.